python/GafferUI/NumericWidget.py

```diff
@@ -285,6 +285,14 @@ def __setValueInternal( self, value, reason ) :
 			self.__numericType = numericType
 			self._qtWidget().setValidator( _ExpressionValidator( self._qtWidget(), self.__numericType ) )
 
+		# If the new value and the current value are the same, don't set the value.
+		# We compare the text representations to account for floating point text formatting.
+		try :
+			if self.getText() and NumericWidget.valueToString( value ) == str( self.getValue() ) :
```

Comment:
We still have a failure in this case :

1. Create a Plane, set `dimensions` to 100.
2. Place cursor to right of the 1.
3. Hit down arrow, and the cursor jumps to the end.

I don't know if it's related, but I don't think we're comparing the right things here. I think we want to compare the value we already have, and the value we'll end up with if we go through with things. That would be `self.getValue() == self.__numericType( valueToString( value ) )`.

Reply:
Yes, that is a better test. Among maybe other things, I was losing the decimal part of floats ending in `.0` which gave a false negative.

Fixed in https://github.com/GafferHQ/gaffer/pull/5370/commits/b8798235ff02f470670eca7716c9e0e06e48020d